src/com/intellij/idea/plugin/hybris/tools/logging/CxLoggerAccess.kt

```diff
@@ -38,29 +40,37 @@ import com.intellij.openapi.project.Project
 import com.intellij.psi.PsiDocumentManager
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
+import java.util.*
 
 @Service(Service.Level.PROJECT)
 class CxLoggerAccess(private val project: Project, private val coroutineScope: CoroutineScope) : Disposable {
     private var fetching: Boolean = false
-    val loggersState = CxLoggersState()
+    private val loggersStates = WeakHashMap<RemoteConnectionSettings, CxLoggersState>()
+    val cxLoggerUtilities: CxLoggerUtilities = CxLoggerUtilities.getInstance(project)
+
 
     val ready: Boolean
         get() = !fetching
 
     val stateInitialized: Boolean
-        get() = loggersState.initialized
+        get() {
+            val server = RemoteConnectionService.getInstance(project).getActiveRemoteConnectionSettings(RemoteConnectionType.Hybris)
+            return loggers(server).initialized
+        }
 
     init {
         with(project.messageBus.connect(this)) {
             subscribe(RemoteConnectionListener.TOPIC, object : RemoteConnectionListener {
-                override fun onActiveHybrisConnectionChanged(remoteConnection: RemoteConnectionSettings) = refresh()
 
-                override fun onActiveSolrConnectionChanged(remoteConnection: RemoteConnectionSettings) = refresh()
+                override fun onHybrisConnectionModified(remoteConnection: RemoteConnectionSettings) = clearState(remoteConnection)
             })
         }
     }
 
-    fun logger(loggerIdentifier: String): CxLoggerModel? = if (!stateInitialized) null else loggersState.get(loggerIdentifier)
+    fun logger(loggerIdentifier: String): CxLoggerModel? {
+        val server = RemoteConnectionService.getInstance(project).getActiveRemoteConnectionSettings(RemoteConnectionType.Hybris)
+        return if (!stateInitialized) null else loggers(server).get(loggerIdentifier)
```

Comment:
flip if statement, positive thinking is simpler from reading.

getActiveRemoteConnectionSettings invoked even when it is not required.